src/utils/index.js

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+/**
+ * Firefox has issues switching quickly between fill style colors, as the CSS color
+ * is fully parsed each time it is set. As a mitigation, provide a class that only
+ * switches the color when it's really needed.
+ */
+export class FastFillStyle {
+  _ctx: CanvasRenderingContext2D;
+  _previousFillColor: string;
+
+  constructor(ctx: CanvasRenderingContext2D) {
+    this._ctx = ctx;
+    this._previousFillColor = '';
+  }
+
+  set(fillStyle: string) {
+    if (fillStyle !== this._previousFillColor) {
+      // This could throw if setCtx wasn't set before calling it. Don't provide an
```

Comment:
This refers to a non-existent function `setCtx`, no? The code below should be fine as we set `_ctx` in the constructor.

Reply:
Yeah, I ended up recreating the instance on every draw call rather than trying to invalidate it.